packages/hash/shared/src/save.ts

```diff
@@ -312,11 +316,64 @@ const updateBlocks = defineOperation(
                 },
               });
             }
+          } else {
+            const draftBlock = getDraftEntityFromEntityId(
+              entityStore.draft,
+              savedEntity.entityId,
+            );
+            if (isBlockEntity(draftBlock)) {
+              const draftChildEntityId =
+                draftBlock?.properties.entity?.entityId;
+              const draftChildEntityWithDraftId = getDraftEntityFromEntityId(
+                entityStore.draft,
+                draftChildEntityId,
+              );
+
+              if (draftChildEntityWithDraftId && draftChildEntityId) {
+                const { draftId: _, ...draftChildEntity } =
+                  draftChildEntityWithDraftId;
+
+                // Check if a block's data has been updated
+                if (
+                  draftChildEntity &&
+                  !isEqual(
+                    draftChildEntity.properties,
+                    savedChildEntity.properties,
+                  )
+                ) {
+                  // If the block's data was updated and the entityId changed
+                  // then the block data was swapped. Push swapBlockData action
+                  if (draftChildEntity.entityId !== savedChildEntity.entityId) {
+                    updates.push({
+                      swapBlockData: {
+                        entityId: savedEntity.entityId,
+                        accountId: savedEntity.accountId,
+                        newEntityAccountId: draftChildEntity.accountId!,
+                        newEntityEntityId: draftChildEntity.entityId!,
+                      },
+                    });
```

Comment:
Is the parent condition necessary for this?

Reply:
Right now it isn't. It was when I was trying to check for updates to non text-entities. the draftId present in draftChildEntity was interfering with the check, so needed to remove that. 
Will remove the conditional, since I don't check for updates in this PR.

Reply:
Updated in https://github.com/hashintel/hash/pull/506/commits/49fa6ec92c043ae0bf450123e317faa4d1390df7